### PR TITLE
feat #443: add 'show this person' function to hole

### PR DESCRIPTION
**Pull request by AInfinity-LilacDream**

### CODE REVIEW

**Comment by AInfinity-LilacDream** [comment]
this function seems to conflict with current "only show dz" function. if you consider that they should co-exist, this pr may solve the problem. or, you may delete the "only show dz" function.

**Comment by HydrogenC** [comment]
Could we unify these two functions? I am saying that we could delete the legacy stuff for `Only DZ`, instead we can just set `selectedPerson=NAME OF DZ` to provide the same functionality in a unified way.

**Comment by w568w** [comment]
> Could we unify these two functions? I am saying that we could delete the legacy stuff for `Only DZ`, instead we can just set `selectedPerson=NAME OF DZ` to provide the same functionality in a unified way.

+1 for this. I think it is the preferred way.

File: lib/l10n/intl_en.arb [hunk 1]
```diff
@@ -544,5 +544,6 @@
   "proxy_setting_input_hint": "e.g. 127.0.0.1:1234 (Leave blank to disable)",
   "proxy_setting_set_successfully": "Proxy settings saved, restart the app to take effect.",
   "use_webvpn_title": "Use WebVPN proxy",
-  "use_webvpn_description": "When enabled, certain functions could be accessed directly without Fudan LAN (restart required)"
+  "use_webvpn_description": "When enabled, certain functions could be accessed directly without Fudan LAN (restart required)",
+  "show_this_person": "Show this person"
```
Comment (w568w) [inline lib/l10n/intl_en.arb:548]: Suggestion: change to `"This user only"`.

File: lib/l10n/intl_ja.arb [hunk 2]
```diff
@@ -526,5 +526,6 @@
   "proxy_setting_input_hint": "例：127.0.0.1:1234（空白のままにしてプロキシを無効にする）",
   "proxy_setting_set_successfully": "プロキシ設定が保存されました。再起動すると有効になります。",
   "use_webvpn_title": "WebVPN を自動的に使う",
-  "use_webvpn_description": "有効時、一部の機能はFudan LANなしで、直接に連接するのができます (再起動が必要)"
+  "use_webvpn_description": "有効時、一部の機能はFudan LANなしで、直接に連接するのができます (再起動が必要)",
+  "show_this_person": "この人を見てください"
```
Comment (w568w) [inline lib/l10n/intl_ja.arb:530]: Suggestion: change to `"この人だけ"`.

File: lib/widget/forum/forum_widgets.dart
Comment (w568w) [inline lib/widget/forum/forum_widgets.dart:1]: Please revert nonsense style changes (e.g. delete/add an empty line) before submitting a PR, if possible.

File: lib/page/forum/hole_detail.dart [hunk 3]
```diff
@@ -1190,7 +1216,9 @@ sealed class RenderModel {}
 class Normal extends RenderModel {
   OTHole hole;
   bool? isFavored, isSubscribed;
-  bool onlyShowDZ = false;
+  bool onlyShowSelectedPerson = false;
+
+  String? selectedPerson;
```
Comment (w568w) [inline lib/page/forum/hole_detail.dart:1221]: As `selectedPerson` can be `null`, why not simply use `selectedPerson == null` to represent the state that `onlyShowSelectedPerson = false`? So you can remove the variable `onlyShowSelectedPerson`.

**Review by w568w — CHANGES_REQUESTED** [review]
LGTM. Minor changes required.

File: lib/page/forum/hole_detail.dart (hunk 3 above)
Comment (AInfinity-LilacDream) [reply lib/page/forum/hole_detail.dart]: Sorry for the unnecessary codes.
Too noob, trying to improve my coding ability.
Wanna assign a couple of more issues in the future.

**Review by w568w — CHANGES_REQUESTED** [review]
You haven't dealt with *every* change request yet. Please tackle all of them before re-request the review.

File: lib/page/forum/hole_detail.dart (hunk 3 above)
Comment (w568w) [reply lib/page/forum/hole_detail.dart]: > Sorry for the unnecessary codes. Too noob, trying to improve my coding ability. 

Never mind! Everyone begins as a noob :)

By the way, don't panic when someone reviews your code! They are totally fine, and **the person pointing out the problem is NOT trying to criticize you (or doubt your abilities)**. We are working together solely to make the project better.

**A better PR deserves a more detailed review.** PRs with no comments are either of little value (e.g., fixing a single obvious bug or merely formatting code) or have already been re-submitted after multiple internal discussions.

Comment (w568w) [reply lib/page/forum/hole_detail.dart]: Also, if you resolved a change request, use `Resolve conversation` button below to close it.

**Review by w568w — APPROVED** [review]
LGTM!

**Pull request merged**